concepts/infrastructure-reconciliation/README.md

```diff
@@ -1,41 +1,172 @@
+The three new operators would need to be introduced:
+- Gardener Cluster Manager
+- External Cluster Manager
+- Subscription Manager
+
+Each of those components has a clear set of responsibilities that together handle the flow that creates a new, complete, Kyma-ready  Kubernetes cluster.
+
+Along the operators, there are four new Custom Resources, that serve as an API for the components:
+- Gardener Cluster CR
+- External Cluster CR
+- SubscriptionRequest CR
+
+### Cluster Managers and the Cluster CRs
+
+The Cluster Managers' main responsibility is to manage the respective Cluster CRs, this is tied to both their provisioning and providing an access to the created K8S clusters.
+
+There will be two resources, first for the Gardener-based, managed clusters:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: GardenerCluster
+metadata:
+	name: managed-cluster
+	namespace: default
+spec:
+	core:
+		purpose: "Production"
+		region: "eu-central-1"
+	extensions:
+		shootNetworkingFilterDisabled: true
+	oidcConfig:
+		clientID: "asdf"
+		groupsClaim: "asdf"
+		issuerURL: "asdf"
+		signingAlgs: ["adsf"]
+		usernameClaim: "asdf"
+		usernamePrefix: "asdf"
+	infrastructure:
+		machineType: "m5.2xlarge"
+		autoScalerMin: 5
+		autoScalerMax: 5
+		provider: aws
+		numberOfZones: 3
+	subscriptionSelector: "globalaccount:abcd-0001" || "shared"
+	kubeconfigs:
+	- secretRef: 
+		name: managed-cluster-admin-kubeconfig
+		namespace: default
+	  clusterRole:
+		rules:
+		- apiGroups: ["*"]
+		  resources: ["*"]
+		  verbs: ["*"]
+	administrators: ["user.id@sap.com", "other.id@sap.com"]
+```
+
+This contains a minimal set of values that are required to provision a new Gardener cluster. The resource also contains the `administrators` and `subscriptionSelector` fields that are required for the Binding creation and requesting a hyperscaler subscription respectively.
+
+And the second one for BYOC model:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: ExternalCluster
+metadata:
+	name: external-cluster
+	namespace: default
+spec:
+	adminKubeconfig: LS0tV1...
+	kubeconfigs:
+	- secretRef: external-cluster-admin-kubeconfig
+	  clusterRole:
+		rules:
+		- apiGroups: ["*"]
+		  resources: ["*"]
+		  verbs: ["*"]
+```
+
+This is the simpler one, as there is no cluster provisioning happening in this flow most of the parameters can be dropped.
+
+### Issuing kubeconfigs
+
+As stated before, the separation of kubeconfigs and limiting access is needed to have a healthy, secure environment. Cluster Managers will provide a possibility to create RBAC-based kubeconfigs in both BYOC and Gardener clusters. This will provide operators with the required access to perform any necessary actions.
+
+The kubeconfigs for specific Kubernetes clusters will be fetched (both from Vault or Gardener) in the runtime and stored only in the components' memory for the limited time needed to issue another (constrained) kubeconfig for a given cluster.
```

Comment:
Gardener Cluster Manager must also need to take care of rotating each shoot's master kubeconfig. I see two options:

1. Keep using the static admin token kubeconfigs for each shoot, and implement configurable time based rotation for them as part of the `GardenerCluster` reconciliation following [this guide](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_credentials_rotation.md#kubeconfig) 
2. Leverate [shoots/adminkubeconfig](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_access.md#shootsadminkubeconfig-subresource) subresource to obtain a  short-living kubeconfig having cluster-admin privileges, when needed (e.g. at the beginning of a `GardenerCluster` reconciliation, when work to be done is detected). Make sure to never persist this kubeconfig in any way, i.e. put to Secrets on KCP.

Reply:
@ebensom I am wondering, why shouldn't we put that short-living admin kubeconfig into the KCP secret? The issue is that the Gardener Cluster Manager will not be able to determine whether there is any work related to the Kyma installation to be done, thus we won't know when to issue such a kubeconfig.

I would rather say using the shoots/adminkubeconfig subresource is better than having the static ones as we will not need to worry about the rotation, we will just need to make sure there are new ones issued accordingly so that the continuity of operations is preserved. And in case this somehow leaks, it will be invalidated with time, not with a rotation.

Reply:
This is the issue, to be honest, I am not sure whether the Roles required by the Module Operator will differ vastly from the admin kubeconfig. In that case, we would create an effective duplicate of the original kubeconfig, and in the end, is there a significant difference between using a newly issued one (with admin-like privileges) and the one from the Gardener?

Since the security assessment, the design has changed a bit. There will be no 3rd party components using those kubeconfigs, and the operators for the specific modules were migrated to the SKR. This leaves only the Module Manager and Lifecycle Manager using those. For the Compass Manager, we will have a separate RBAC kubeconfig for sure though.

Reply:
In that case, no. there isn't much of a difference in terms of granted permissions.
The Gardener issued kubeconfig uses client certificate authentication, where the certificate has the short expiration date, and the identity of the certificate (DN) is the name of the service account which requested the kubeconfig (e.g. compass-provisioner).
In the case of the proposal's RBAC kubeconfig, it uses token issued for a K8S service account created on the cluster by the cluster operator, which would be expired and deleted from the cluster after some time.
Ultimately they all grant cluster-admin permissions to a technical user identity, so no big difference from result point of view.